src/runtime/node/net/internal/server.ts

```diff
@@ -0,0 +1,47 @@
+import type * as net from "node:net";
+import { EventEmitter } from "node:events";
+
+// Docs: https://nodejs.org/api/net.html#net_class_net_server
+export class Server extends EventEmitter implements net.Server {
+  readonly __unenv__ = true;
+
+  maxConnections: number = 1;
+  connections: number = 0;
+  readonly listening: boolean = false;
+
+  constructor(
+    arg1?: net.ServerOpts | ((socket: net.Socket) => void),
+    arg2?: (socket: net.Socket) => void,
+  ) {
+    super();
+  }
+
+  listen(): this {
+    return this;
```

Comment:
Maybe we should just call notImplemented here and throw error?

Reply:
Added fail: d364dec83664f56da994f275f90c02f40b5a6ed8

My main intention was to not fail if any lib is using listen for exmaplte to test a random port. But being explicit is probably bettter.

Reply:
If this doesn't throw an error, probably their own code will fail at some point, and they'd assume something similar to the original issue that this PR fixes.